src/hooks/screen/useChannelHook.ts

```diff
@@ -3,41 +3,45 @@ import {PostNotificationChannelList} from '../../../types/database/schema/PostNo
 
 function useChannelHook() {
   const determinePostType = (data: PostNotificationChannelList): BaseChannelItemTypeProps => {
-    const isOwningPost = data?.rawJson?.isOwnPost;
+    const isOwnSignedPost =
+      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
+    const isOwnAnonymousPost =
+      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new variables `isOwnSignedPost` and `isOwnAnonymousPost` are being checked for truthiness before they are used. This is a good practice as it prevents potential runtime errors due to undefined or null values.

However, the check for `data?.channelType === 'POST_NOTIFICATION'` and `data?.channelType === 'ANON_POST_NOTIFICATION'` seems redundant in these lines. If `isOwnSignedPost` and `isOwnAnonymousPost` are properties of `rawJson`, then their existence already implies that the channel type is either 'POST_NOTIFICATION' or 'ANON_POST_NOTIFICATION'. Unless there's a case where these properties can exist outside of these channel types, you might want to remove this redundancy.
```diff
-    const isOwnSignedPost =
-      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
-    const isOwnAnonymousPost =
-      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';
+    const isOwnSignedPost = data?.rawJson?.isOwnSignedPost;
+    const isOwnAnonymousPost = data?.rawJson?.isOwnAnonymousPost;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->